Expensify/App
# Fix incorrect unread count notification show up

### Details

Pusher event coming but  `LastReadSequenceNumbers` isn't ready, `LastReadSequenceNumbers` if filled by `fetchAllReports`.  This will cause  the application to show big number of unread count which is actually a `maxSequenceNumber` of the report 

### Fixed Issues

$ https://github.com/Expensify/App/issues/6591
$ https://github.com/Expensify/App/issues/7914

### Tests

1. Need firefox/chrome developer tools to do the test, doing the test in web platform
2. Open any chat room 
3. In network tab of developers tools, configure throttling speed to be 20kbps.
4. Refresh browser tab
5. In developer tools > network tab, observe push_authetication request and wait until it is completed
6. Right after the push_authenticate completed, send message to a room (other than active/currently visited room) from other device
7. Verify unread count notification is not a big number


### QA Steps


1. Need firefox/chrome developer tools to do the test, doing the test in web platform
2. Open any chat room 
3. In network tab of developers tools, configure throttling speed to be 20kbps.
4. Refresh browser tab
5. In developer tools > network tab, observe push_authetication request and wait until it is completed
6. Right after the push_authenticate completed, send message to a room (other than active/currently visited room) from other device
7. Verify unread count notification is not a big number

For IOS:
1. User A Sign out if he is already sign in
2. Sign in 
3. Right after sign in immediately go to home screen/background the application
4. From other user (User B) send a message to user A
5. In IOS of User A: Verify application icon badge count (unread count) in home screen of IOS doesn't display big number and display the unread count properly. 


### Tested On






### Screenshots


#### Web

##### Before


https://user-images.githubusercontent.com/1313124/152940360-079fbaa4-2a7e-4911-abee-48aac8ea62fe.mp4




##### After

https://user-images.githubusercontent.com/1313124/152937601-50ac7b20-bf6a-4056-8a72-1595bb4e07b7.mp4


#### Mobile Web

N/A

#### Desktop


Before:


https://user-images.githubusercontent.com/1313124/154592731-1fde61ac-f879-446c-9bcd-35c3b7f2c1ea.mp4


After:

https://user-images.githubusercontent.com/1313124/154592339-9d320431-735b-47eb-aa63-cb2c80b9a748.mp4

#### iOS

Will try to add video later.

#### Android

N/A
